server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -729,13 +737,16 @@ private static void parseNonDynamicArray(
             }
         }
 
-        // In synthetic source, if any array element requires storing its source as-is, it takes precedence over
-        // elements from regular source loading that are then skipped from the synthesized array source.
-        // To prevent this, we track that parsing sub-context is within array scope.
-        context = context.maybeCloneForArray(mapper);
+        if (parsesArrayValue(mapper) == false && supportStoringArrayOffsets == false) {
```

Comment:
`maybeCloneForArray` only cares about objects, is this check relevant?

Reply:
Yes, turns out this wasn't needed at all.

Reply:
It was there, because of how this PR worked initially, but currently not needed anymore.